test/e2e/testdata/cmp-gitsshcreds/plugin.yaml

```diff
@@ -0,0 +1,11 @@
+apiVersion: argoproj.io/v1alpha1
+kind: ConfigManagementPlugin
+metadata:
+  name: cmp-gitcredstemplate
+spec:
+  version: v1.0
+  generate:
+    command: ["sh", "generate.sh"]
+  discover:
+    fileName: "subdir/s*.yaml"
+  provideGitCreds: true
```

Comment:
Should we have another test to prove that we don't send creds when this flag is false?

Reply:
I can do that, I can write a script to confirm that specific env vars are empty.

Reply:
new integration test added for an SSH clone failing in the sidecar